Add guide for customizing Helm chart values (#1556)
webrecorder/browsertrix: docs/deploy/helm-customization.md

# Helm Chart Customization

Local and production deployments alike can be customized by modifying the ``chart/values.yaml`` Helm chart file or a local override. For more on using local overrides, see the [Local Deployment Guide](local.md). The remainder of this guide covers some of the customization options available in the Helm chart.

## Default Organization

> The ``default_org`` setting is used to specify the name for the default organization created in a Browsertrix Cloud deployment. A slug will be auto-generated based on this value and can be modified in Org Settings within the application.

Reviewer: I know we haven't officially renamed it yet, but we're basically going to at this point.  Have been referring to the app as Browsertrix everywhere else recently.

Also adds a link to the user guide page.

```suggestion
The ``default_org`` setting is used to specify the name for the default organization created in a Browsertrix deployment. A slug will be auto-generated based on this value and can be modified in [Org Settings](../user-guide/org-settings.md) within the application.
```

## Superuser

> The ``superuser`` setting is used to set the username and password for a deployment's superuser. If ``password`` is left blank, the application will auto-generate a secret password for the superuser.

Author: ```suggestion
The ``superuser`` setting is used to set the username and password for a deployment's superuser. If ``password`` is left blank, the application will auto-generate a secure password for the superuser.
```

## Crawler Channels

> The ``crawler_channels`` setting is used to specify the crawler versions available to users via dropdown menus in workflows and browser profiles. Each crawler channel has an id as well as a Docker image tag. These channels are modifiable with the restriction that there must always be one channel with the id ``default``. By default this is the only channel available on deployments:

Reviewer: Adds link to user guide section, mentions specific gui setting by name.
```suggestion
The ``crawler_channels`` setting is used to specify the [_Crawler Release Channel_](../user-guide/workflow-setup.md#crawler-release-channel) option available to users via dropdown menus in workflows and browser profiles. Each crawler channel has an id and a Docker image tag. These channels are modifiable with the restriction that there must always be one channel with the id ``default``. By default this is the only channel available on deployments:
```

```yaml
crawler_channels:
  - id: default
    image: "docker.io/webrecorder/browsertrix-crawler:latest"
```

This can be extended with additional channels. For example, here is what the value would look like adding a new x.y.z release of Browsertrix Crawler with the id ``testing``:

```yaml
crawler_channels:
  - id: default
    image: "docker.io/webrecorder/browsertrix-crawler:latest"
  - id: testing
    image: "docker.io/webrecorder/browsertrix-crawler:x.y.z"
```

## Storage

> The ``storage`` setting is used to specify primary and replica storage for a Browsertrix Cloud deployment. All configured storage options must be S3-compatible buckets. At minimum, there must be one configured storage option, as can be seen in the default configuration:

Reviewer: ```suggestion
The ``storage`` setting is used to specify primary and replica storage for a Browsertrix deployment. All configured storage options must be S3-compatible buckets. At minimum, there must be one configured storage option, as can be seen in the default configuration:
```

For example, here is what a storage configuration with two replica locations, one in another bucket on the same Minio S3 service as primary storage as well as another in an external S3 provider:

```yaml
storages:
  - name: "default"
    type: "s3"
    access_key: "ADMIN"
    secret_key: "PASSW0RD"
    bucket_name: *local_bucket_name

    endpoint_url: "http://local-minio.default:9000/"
    is_default_primary: True

  - name: "replica-0"
    type: "s3"
    access_key: "ADMIN"
    secret_key: "PASSW0RD"
    bucket_name: "replica-0"

    endpoint_url: "http://local-minio.default:9000/"
    is_default_replica: True
  
  - name: "replica-1"
    type: "s3"
    access_key: "accesskey"
    secret_key: "secret"
    bucket_name: "replica-1"

    endpoint_url: "http://s3provider.example.com"
```

## Email / SMTP Server

> Browsertrix Cloud sends user invitations, password resets, background job failure notifications, and other important messages via email. The ``email`` setting can be used to configure the SMTP server used to send emails. To avoid email messages from Browsertrix Cloud being flagged as spam, be sure to use the same domain for ``sender_email`` and ``reply_to_email``.

Reviewer: ```suggestion
Browsertrix sends user invitations, password resets, background job failure notifications, and other important messages via email. The ``email`` setting can be used to configure the SMTP server used to send emails. To avoid email messages from Browsertrix being flagged as spam, be sure to use the same domain for ``sender_email`` and ``reply_to_email``.
```

## Signing WACZ files

> Browsertrix Cloud has the ability to cryptographically sign WACZ files with [Authsign](https://github.com/webrecorder/authsign). The ``signer`` setting can be used to enable this feature and configure Authsign.

Reviewer: ```suggestion
Browsertrix has the ability to cryptographically sign WACZ files with [Authsign](https://github.com/webrecorder/authsign). The ``signer`` setting can be used to enable this feature and configure Authsign.
```
